

<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>Welcome to pyTEnrich’s documentation! &mdash; pyTEnrich 0.3 documentation</title>
  

  
  <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="_static/pygments.css" type="text/css" />

  
  
  
  

  
  <!--[if lt IE 9]>
    <script src="_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
    
      <script type="text/javascript" id="documentation_options" data-url_root="./" src="_static/documentation_options.js"></script>
        <script src="_static/jquery.js"></script>
        <script src="_static/underscore.js"></script>
        <script src="_static/doctools.js"></script>
        <script async="async" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.7/latest.js?config=TeX-AMS-MML_HTMLorMML"></script>
    
    <script type="text/javascript" src="_static/js/theme.js"></script>

    
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="Installation" href="usage/installation.html" /> 
</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">
    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
          

          
            <a href="#" class="icon icon-home" alt="Documentation Home"> pyTEnrich
          

          
          </a>

          
            
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        
        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <p class="caption"><span class="caption-text">Contents:</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="usage/installation.html">Installation</a></li>
<li class="toctree-l1"><a class="reference internal" href="usage/execution.html">Execution</a></li>
<li class="toctree-l1"><a class="reference internal" href="usage/otherspecies.html">Working with other species</a></li>
</ul>

            
          
        </div>
        
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="#">pyTEnrich</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content">
        
          















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="#" class="icon icon-home"></a> &raquo;</li>
        
      <li>Welcome to pyTEnrich’s documentation!</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
            
            <a href="_sources/index.rst.txt" rel="nofollow"> View page source</a>
          
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="section" id="welcome-to-pytenrich-s-documentation">
<h1>Welcome to pyTEnrich’s documentation!<a class="headerlink" href="#welcome-to-pytenrich-s-documentation" title="Permalink to this headline">¶</a></h1>
<p>A code for to compute statistical enrichment of transposable elements
on a group of bed files. Code written by Alexandre Coudray from the
laboratory of Virology and Genetics at the EPFL in 2019.</p>
<div class="toctree-wrapper compound">
<p class="caption"><span class="caption-text">Contents:</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="usage/installation.html">Installation</a></li>
<li class="toctree-l1"><a class="reference internal" href="usage/execution.html">Execution</a></li>
<li class="toctree-l1"><a class="reference internal" href="usage/otherspecies.html">Working with other species</a></li>
</ul>
</div>
</div>
<div class="section" id="overview-of-the-methods">
<h1>Overview of the methods<a class="headerlink" href="#overview-of-the-methods" title="Permalink to this headline">¶</a></h1>
<p>To compute the statistical enrichment, pyTEnrich compare the overlap between input bed files and transposable elements families. pyTEnrich uses a binomial model to compare the observed overlap to the expected one, given a probability computed from the genome occupancy of TE families. The goal is to compute a fold change and a p-value for the overlap observed between each TE families and a group of input bed files.</p>
<p><strong>Example</strong></p>
<p>We consider a small genome with one gene composed of two exons, and two TE families. There is 3 ChIP-seq peaks detected :</p>
<img alt="_images/fig1.png" src="_images/fig1.png" />
<p><strong>Step 1 : Compute genome occupancy</strong></p>
<p>First, pyTEnrich compute genome occupancy for TE families and input bed files.</p>
<img alt="_images/fig2.jpg" src="_images/fig2.jpg" />
<p>Note that TE genome occupancy are pre-computed for the provided TE database. If another TE database is given, or if a genome subset is provided (explained below), it will be re-computed (takes a few minutes).</p>
<p><strong>Step 2 : Intersect TE and Input bed files and count overlap</strong></p>
<p>Using Bedtools intersect (link to website), we compute a stringent overlap between input bed files and TE database. The observed overlap are counted to be compared with the expectations.</p>
<img alt="_images/fig3.jpg" src="_images/fig3.jpg" />
<p><strong>Step 3 : Compute the enrichment of TE subfamily / family</strong></p>
<p>For each input bed files, and enrichment is performed using a binomial test. The two possible outcome according to our model is : either TE overlap with peak (success) or no overlap (failure) with a probability p. p was calculated using genome occupancy of each TE subfam. The number of trials N corresponds to the number of peaks in input bed file.</p>
<p>The binomial probability of having k successes from n trials is given by</p>
<div class="math notranslate nohighlight">
\[P(B = k) = \binom{n}{k} p^k (1 - p)^{n - k}\]</div>
<p>We can calculate the probability to have at least k success by suming up probabilities, from k success to n success. As often the number of success is on the low edge, we prefer to compute the inverse probability :</p>
<div class="math notranslate nohighlight">
\[P(B &gt;= k) = 1 - P(B &lt; k) = 1 - \sum_{i=0}^{k-1} \binom{n}{i} p^i (1 - p)^{n - i}\]</div>
<p>This probability is our p-value of having at least k success, given a probability p for the overlap, and n trials. The p-values obtained above are then adjusted with the Benjamin-Hochsberg method to correct for multiple testing.</p>
<p><strong>Enrichment analysis example on Imbeault et al. dataset</strong></p>
<p>Using the abovementionned approach, TE enrichment analysis was performed for 321 ChIP-exo from <a class="reference external" href="https://www.nature.com/articles/nature21683">Imbeault et al. 2018 dataset</a>.</p>
<img alt="_images/fig4.jpg" src="_images/fig4.jpg" />
<p><strong>Optional parameter : Genome subset</strong></p>
<p>A genome subset can be given as an optional parameter to compute the enrichment only on a subset of the genome. For instance, we might interest ourself only on the neighborhood of transcription start sites (TSS). Therefore, a bed file containing these regions can be provided through the <strong>–genome_subset</strong> parameter.</p>
<img alt="_images/fig5.jpg" src="_images/fig5.jpg" />
</div>


           </div>
           
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="usage/installation.html" class="btn btn-neutral float-right" title="Installation" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right"></span></a>
      
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>
        
        &copy; Copyright 2020, Laboratory of Virology and Genetics

    </p>
  </div>
    
    
    
    Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a
    
    <a href="https://github.com/rtfd/sphinx_rtd_theme">theme</a>
    
    provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  

  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script>

  
  
    
   

</body>
</html>